diff -r -c gmic-qt/src/gmic_qt.cpp gmic-qt-new/src/gmic_qt.cpp
*** gmic-qt/src/gmic_qt.cpp	Sun Oct 14 12:36:05 2018
--- gmic-qt-new/src/gmic_qt.cpp	Mon Oct 15 17:42:48 2018
***************
*** 68,73 ****
--- 68,90 ----
    SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX | SEM_NOOPENFILEERRORBOX);
  #endif
  
+   QString appDir;
+   {
+     char exname[1024] = {0};
+     // get the path where the executable is stored
+     uint32_t size = sizeof(exname);
+     if (_NSGetExecutablePath(exname, &size) == 0)
+       printf("executable path is %s\n", exname);
+     else
+       printf("buffer too small; need size %u\n", size);
+ 
+     QDir dir(exname);
+     appDir = dir.absolutePath();
+     appDir += "GMIC/plugins";
+   }
+   QCoreApplication::addLibraryPath(appDir);
+   printf("QCoreApplication::addLibraryPath(%s)\n", appDir.data());
+ 
    QApplication app(dummy_argc, dummy_argv);
    app.setWindowIcon(QIcon(":resources/gmic_hat.png"));
    QCoreApplication::setOrganizationName(GMIC_QT_ORGANISATION_NAME);
