diff -r -c gmic-qt/src/gmic_qt.cpp gmic-qt-new/src/gmic_qt.cpp
*** gmic-qt/src/gmic_qt.cpp	Sun Oct 14 12:36:05 2018
--- gmic-qt-new/src/gmic_qt.cpp	Mon Oct 15 13:29:20 2018
***************
*** 68,73 ****
--- 68,78 ----
    SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX | SEM_NOOPENFILEERRORBOX);
  #endif
  
+   QDir dir(QCoreApplication::applicationDirPath());
+   QString path = dir.absolutePath();
+   path += "GMIC/plugins";
+   QCoreApplication::addLibraryPath(path);
+ 
    QApplication app(dummy_argc, dummy_argv);
    app.setWindowIcon(QIcon(":resources/gmic_hat.png"));
    QCoreApplication::setOrganizationName(GMIC_QT_ORGANISATION_NAME);
